src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1895,15 +1917,25 @@ public override BoundNode VisitConversion(BoundConversion node)
                         break;
                 }
 
-                if (node.ExplicitCastInCode &&
-                    !node.IsExplicitlyNullable &&
-                    targetType.IsReferenceType &&
-                    (operandType?.IsNullable == true || (operandType is null && operand.IsLiteralNullOrDefault())))
+                if (node.ExplicitCastInCode && !node.IsExplicitlyNullable)
                 {
-                    // PROTOTYPE(NullableReferenceTypes): Should not report warning for explicit
-                    // user-defined conversion if the operator is defined from nullable to
-                    // non-nullable. See StaticNullChecking.ExplicitCast_UserDefined.
-                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_ConvertingNullableToNonNullable, node.Syntax);
+                    bool reportNullable = false;
+                    if (targetType.IsReferenceType && IsUnconstrainedTypeParameter(operandType?.TypeSymbol))
```

Comment:
>IsUnconstrainedTypeParameter(operandType?.TypeSymbol) [](start = 54, length = 53)

Are we tracking flow state for expressions like this? Is it possible that we know that it is not null at this point? For example, there was a null check before.

Reply:
We are not tracking nullability of unconstrained type parameters yet.

Reply:
> We are not tracking nullability of unconstrained type parameters yet.

Consider adding a test for the scenario and a PRPOTOTYPE comment if necessary.

---
In reply to: [178483878](https://github.com/dotnet/roslyn/pull/25733#discussion_r178483878) [](ancestors = 178483878)

Reply:
Added `TrackUnconstrainedTypeParameter_ExplicitCast` test.

---
In reply to: [178484817](https://github.com/dotnet/roslyn/pull/25733#discussion_r178484817) [](ancestors = 178484817,178483878)